irepela/homework_6/unpack.py

```diff
@@ -0,0 +1,50 @@
+def unpack_while_loop(items: list) -> list:
+    """
+      Flattens list with up to 3 levels of nesting
+
+      Args:
+          items: list of items with up to 3 levels of nesting
+
+      Returns:
+          list: flattened list
+    """
+    unpacked = []
+    copied_list = list(reversed(items))
+    while len(copied_list) > 0:
+        next_item = copied_list.pop()
+        if isinstance(next_item, list):
+            nested_list_1 = list(reversed(next_item))
+            while len(nested_list_1) > 0:
+                next_item = nested_list_1.pop()
+                if isinstance(next_item, list):
+                    nested_list_2 = list(reversed(next_item))
+                    while len(nested_list_2) > 0:
+                        next_item = nested_list_2.pop()
+                        if isinstance(next_item, list):
+                            unpacked.extend(next_item)
+                        else:
+                            unpacked.append(next_item)
+                else:
+                    unpacked.append(next_item)
+        else:
+            unpacked.append(next_item)
+    return unpacked
```

Comment:
you can not predict depth of nested structure

so you need to manipulate with queue

get item from queue, return items to queue, repeat